plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/PartitionProjection.java

```diff
@@ -52,6 +53,31 @@ public PartitionProjection(Optional<String> storageLocationTemplate, Map<String,
         this.columnProjections = ImmutableMap.copyOf(requireNonNull(columnProjections, "columnProjections is null"));
     }
 
+    // TODO: support writing partition projection
+    public boolean allowWrite()
+    {
+        for (Projection projection : columnProjections.values()) {
+            // DateProjection may contain a user-defined format that is incompatible with Hive.
+            // Currently, we only support writing partitions in Hive's default date format.
+            // Allowing custom date formats in partition projection can lead to inconsistencies
+            // between reading and writing of the projection partition.
+            // To ensure compatibility and avoid incorrect query results, disable writing for now.
+            if (projection instanceof DateProjection dateProjection) {
+                return dateProjection.allowWrite();
```

Comment:
As written, this logic would return the result of `dateProjection.allowWrite()` on the first encountered `DateProjection` and would not check any remaining date projections.